packages/ui-react/src/containers/AccountModal/forms/PersonalDetails.tsx

```diff
@@ -33,6 +33,27 @@ const PersonalDetails = () => {
   const [questionErrors, setQuestionErrors] = useState<Record<string, string>>({});
 
   const fields = useMemo(() => Object.fromEntries(data?.settings.map((item) => [item.key, item]) || []), [data]);
+
+  // if we use the register form and use social media we're redirected to this modal, but it could be the case that we've
+  // already previously registered and filled out the same form, in which case we need to skip this modal entirely
+  const isPreviouslyRegistered = useMemo(() => {
+    if (!data) return false;
+
+    const isEmpty = (item: CleengCaptureField | CleengCaptureQuestionField) => {
+      if (item.answer === null) return true;
+      if (typeof item.answer === 'string') return !item.answer.trim();
+      if (typeof item.answer === 'object') return Object.values(item.answer).every((value) => !value || !value.trim());
+    };
+
+    const enabledFields = data.settings.filter((item) => item.enabled);
+
+    const hasSomeFilled = enabledFields.some((item) => !isEmpty(item));
+
+    const hasNoneRequiredEmpty = !enabledFields.some((item) => item.required && isEmpty(item));
+
+    return hasSomeFilled && hasNoneRequiredEmpty;
+  }, [data]);
```

Comment:
This logic is way too specific for this container. Can't we determine this using the `getCaptureStatus` method?

Reply:
Hmmm, then I'm afraid the social login/register user flow isn't quite stable.

The `shouldCaptureBeDisplayed` value reflects whether the personal details form should be shown due to an update in the capture fields or the user hasn't filled in anything yet. The Cleeng API determines this for us.

The JWP service will always return true in this case:

https://github.com/jwplayer/ott-web-app/blob/d6c720271040ee67606e5e678c0179112474bd3a/packages/common/src/services/integrations/jwp/JWPAccountService.ts#L396-L399

Does it make more sense to do the check here and leverage the existing flow?

The reason I bring this up is because we are trying to extract most business logic to service/controllers, and this change re-adds some integration-specific logic into the UI.

Reply:
That's a good suggestion, I'll look into it right away.

Reply:
@ChristiaanScheermeijer please take a look at my last commit:
https://github.com/jwplayer/ott-web-app/pull/499/commits/a9d3630ed790d482bb038f1eecfd74b3ee68fa9e